File: test/data/phenology/crop2ml/algo/f90/PhyllochronModule.f90
    !use crop2mlModules
        IF (choosePhyllUse .EQ. 'Default') THEN
            IF (leafNumber < ldecr) THEN
                phyllochron = fixPhyll * pdecr
            ELSE IF ((leafNumber >= ldecr) .AND. (leafNumber < lincr)) THEN
                phyllochron = fixPhyll
            ELSE
                phyllochron = fixPhyll * pincr
            END IF
        END IF
        IF (choosePhyllUse .EQ. 'PTQ') THEN
            gai = MAX(pastMaxAI,gai)
            pastMaxAI = gai
            IF (gai > 0.0) THEN
                phyllochron = phylPTQ1 * ((gai * kl) / (1 - EXP(-kl * gai))) / (ptq + aPTQ)
            ELSE
                phyllochron = phylPTQ1
            END IF
        END IF
        IF (choosePhyllUse == 'Test') THEN

            IF (leafNumber < ldecr) THEN
                phyllochron = p * pdecr
            ELSE IF ((leafNumber >= ldecr) .AND. (leafNumber < lincr)) THEN
                phyllochron = p
            ELSE
                phyllochron = p * pincr
            END IF
        END IF

File: test/data/phenology/crop2ml/algo/cpp/PhyllochronModule.cpp


if (choosePhyllUse == "Default")
{
    if (leafNumber < ldecr)
        phyllochron = fixPhyll * pdecr;
    else if (leafNumber >= ldecr && leafNumber < lincr)
        phyllochron = fixPhyll;
    else
        phyllochron = fixPhyll * pincr;
}
if (choosePhyllUse == "PTQ")
{
    gai = fmax(pastMaxAI, gai);
    pastMaxAI = gai;
    if (gai > 0.0)
        phyllochron = phylPTQ1 * ((gai * kl) / (1 - exp(-kl * gai))) / (ptq + aPTQ);
    else
        phyllochron = phylPTQ1;
}
if (choosePhyllUse == "Test")
{
    if (leafNumber < ldecr)
        phyllochron = p * pdecr;
    else if (leafNumber >= ldecr && leafNumber < lincr)
        phyllochron = p;
    else
        phyllochron = p * pincr;
}
